core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -0,0 +1,242 @@
+import { Node } from 'prosemirror-model';
+import { EditorState, NodeSelection, Plugin, PluginKey, Transaction } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { Subject } from 'rxjs';
+import tippy, { Instance, Props } from 'tippy.js';
+
+import { ComponentRef } from '@angular/core';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { Editor, posToDOMRect } from '@tiptap/core';
+
+import { getNodePosition } from '../../bubble-menu/utils';
+import { AIContentPromptComponent } from '../ai-content-prompt.component';
+import { AI_CONTENT_PROMPT_PLUGIN_KEY } from '../ai-content-prompt.extension';
+import { TIPPY_OPTIONS } from '../utils';
+
+interface AIContentPromptProps {
+    pluginKey: PluginKey;
+    editor: Editor;
+    element: HTMLElement;
+    tippyOptions?: Partial<Props>;
+    component: ComponentRef<AIContentPromptComponent>;
+}
+
+interface PluginState {
+    open: boolean;
+    form: [];
+}
+
+export type AIContentPromptViewProps = AIContentPromptProps & {
+    view: EditorView;
+};
+
+export class AIContentPromptView {
+    public editor: Editor;
+
+    public node: Node;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component?: ComponentRef<AIContentPromptComponent>;
+
+    private $destroy = new Subject<boolean>();
+
+    constructor(props: AIContentPromptViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.element.style.visibility = 'visible';
+        this.pluginKey = pluginKey;
+        this.component = component;
+
+        this.component.instance.formValues.pipe(takeUntil(this.$destroy)).subscribe((data) => {
+            this.editor.commands.updateValue(data);
+        });
+
+        this.component.instance.hide.pipe(takeUntil(this.$destroy)).subscribe(() => {
+            this.editor.commands.closeAIPrompt();
+        });
+
+        this.editor.on('focus', this.focusHandler);
+
+        document.body.addEventListener('scroll', this.hanlderScroll.bind(this), true);
+
+        document.body.addEventListener('mousedown', this.handleOutsideClick, { capture: true });
```

Comment:
Can we remove this listeners?

Reply:
I reviewed the listeners:

- We can safely remove the 'scroll' listener, as it's not required in this context.
- However, the 'mousedown' listener is essential for closing the content when clicking outside, so it needs to remain.